python/box.tests/test/org/jetbrains/kotlin/python/test/JsTestChecker.kt

```diff
@@ -104,15 +105,25 @@ object PythonTestChecker : AbstractJsTestChecker() {
 
     private fun runPython(scriptPath: String): String {
         val process = Runtime.getRuntime().exec("python3.8 $scriptPath")
-        process.waitFor(10, TimeUnit.SECONDS)
+        val runningTime = measureTimeMillis {
+            process.waitFor(10, TimeUnit.SECONDS)
+        }
+        val runningTimeMessage = "Python running time: $runningTime ms"
+        val exitValue = try {
+            process.exitValue()
+        } catch (e: IllegalThreadStateException) {
+            // If the process hasn't finished yet.
+            process.destroyForcibly()
+            123 // Some non-zero value, doesn't matter what.
+        }
 
-        if (process.exitValue() == 0) {
+        if (exitValue != 0) {
+            val stderrReader = BufferedReader(InputStreamReader(process.errorStream))
+            val stderr = stderrReader.lineSequence().toList().takeLast(10).joinToString("\n")
+            throw PythonExecutionException("$runningTimeMessage\n$stderr")
```

Comment:
I have the following errors sometimes:

![image](https://user-images.githubusercontent.com/20105593/130232066-7a939c42-f476-4402-bc33-202d4d455c00.png)

Could you please take a look? I guess PythonExecutionException should be logged, but it's not.

Reply:
Partially fixed in https://github.com/krzema12/kotlin-python/pull/23/commits/3b2474c683f3f0b1515ecdb213ebf9be0f094d8a. There's some issue with extracting STDERR content, I found it for 18 tests so not much. It happens when the process execution times out, so I'd expect some problems with contacting with it.

Reply:
Now we get:

```
Python running time: 10000 ms
There was a problem when extracting the contents of STDERR. Details: java.io.IOException: Stream closed
org.jetbrains.kotlin.python.test.PythonExecutionException: Python running time: 10000 ms
```